app/controllers/people_controller.rb

```diff
@@ -117,9 +115,28 @@ def create
 
   def update
     @person = Person.find(params[:id])
-    
+    channel_params = params[:person][:channels_attributes].values
+
+    unless channel_params.empty?
+        channel_params.each do |hash|
```

Comment:
Excellent point. I called it hash because I wanted to keep in mind how I needed to manipulate it. I can definitely change that!